go/server.go

```diff
@@ -23,38 +24,32 @@ func init() {
 var (
 	PLAID_CLIENT_ID     = os.Getenv("PLAID_CLIENT_ID")
 	PLAID_SECRET        = os.Getenv("PLAID_SECRET")
-	PLAID_PUBLIC_KEY    = os.Getenv("PLAID_PUBLIC_KEY")
 	PLAID_PRODUCTS      = os.Getenv("PLAID_PRODUCTS")
 	PLAID_COUNTRY_CODES = os.Getenv("PLAID_COUNTRY_CODES")
 	// Parameters used for the OAuth redirect Link flow.
 	//
-	// Set PLAID_OAUTH_REDIRECT_URI to 'http://localhost:8000/oauth-response.html'
+	// Set PLAID_REDIRECT_URI to 'http://localhost:8000/oauth-response.html'
 	// The OAuth redirect flow requires an endpoint on the developer's website
 	// that the bank website should redirect to. You will need to configure
 	// this redirect URI for your client ID through the Plaid developer dashboard
 	// at https://dashboard.plaid.com/team/api.
-	PLAID_OAUTH_REDIRECT_URI = os.Getenv("PLAID_OAUTH_REDIRECT_URI")
-	// Set PLAID_OAUTH_NONCE to a unique identifier such as a UUID for each Link
-	// session. The nonce will be used to re-open Link upon completion of the OAuth
-	// redirect. The nonce must be at least 16 characters long.
-	PLAID_OAUTH_NONCE = os.Getenv("PLAID_OAUTH_NONCE")
+	PLAID_REDIRECT_URI = os.Getenv("PLAID_REDIRECT_URI")
 
 	// Use 'sandbox' to test with fake credentials in Plaid's Sandbox environment
 	// Use `development` to test with real credentials while developing
 	// Use `production` to go live with real users
 	APP_PORT = os.Getenv("APP_PORT")
 )
 
-var clientOptions = plaid.ClientOptions{
-	PLAID_CLIENT_ID,
-	PLAID_SECRET,
-	PLAID_PUBLIC_KEY,
-	plaid.Sandbox, // Available environments are Sandbox, Development, and Production
-	&http.Client{},
+func createClient(environment plaid.Environment) (client *plaid.Client, err error) {
+	return plaid.NewClient(plaid.ClientOptions{
+		PLAID_CLIENT_ID,
+		PLAID_SECRET,
+		environment, // Available environments are Sandbox, Development, and Production
+		&http.Client{},
+	})
 }
 
-var client, err = plaid.NewClient(clientOptions)
```

Comment:
wondering if we should define the functions in `server.go` as methods on a new `plaidClient` struct receiver to avoid having to re-initialize a plaid client every time and avoid the not ideal initialization logic: `var client, err = plaid.NewClient(clientOptions)`

we can use the golang init function https://golang.org/doc/effective_go.html#init to initialize `plaidClient`

this also feels like the implementation we should recommend to our developers perhaps

Reply:
@skansal-plaid To what an extent does introducing a new `plaidClient` struct make understanding how to use `plaid.Client` harder as it is another level of abstraction specific to quickstart/go ?

Reply:
@afh perhaps to avoid the additional abstraction, we could preserve the `var client, err = plaid.NewClient(clientOptions)` as is (or keep these changes out of scope from this PR)?

for the most part, my push is to avoid re-initializing the plaid client multiple times since that feels like something we shouldn't encourage for developers -- curious how we should solve this / if we are ok with re-initialization already. in any case, i will defer to @davidzhanghp